pkg/vulnerability/vulnerability.go

```diff
@@ -0,0 +1,100 @@
+package vulnerability
+
+import (
+	"bufio"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/utils"
+)
+
+const (
+	DefaultIgnoreFile = ".trivyignore"
+)
+
+type Client struct {
+	dbc db.Operations
+}
+
+func NewClient() Client {
+	return Client{
+		dbc: db.Config{},
+	}
+}
+
+func (c Client) FillAndFilter(vulns []types.DetectedVulnerability, severities []dbTypes.Severity,
+	ignoreUnfixed bool, ignoreFile string, light bool) []types.DetectedVulnerability {
+	var err error
+	var severity dbTypes.Severity
+
+	ignoredIDs := getIgnoredIDs(ignoreFile)
+	var vulnerabilities []types.DetectedVulnerability
+	for _, vuln := range vulns {
+		var vulnerability dbTypes.Vulnerability
+		if light {
+			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
+			vulnerability.Severity = severity.String()
```

Comment:
What if err != nil?

Reply:
In that case, the error is handled by line 46 - 49. The error would be displayed, but it wouldn't stop the loop.